### PR TITLE
Updated some golden image tests for M2/M3

### PR DESCRIPTION
**Pull request by HansMuller**
Updated some of the golden image tests that were temporarily changed in https://github.com/flutter/flutter/pull/128914 to have M2 and M3 versions. 

Updated the linear_gradient_0 example to M3 (which will require updating its golden image as well).

More info in https://github.com/flutter/flutter/issues/127064

### CODE REVIEW

File: packages/flutter/test/rendering/localized_fonts_test.dart [hunk 1]
```diff
@@ -55,13 +55,59 @@ void main() {
...
   );
 
   testWidgets(
-    'Text with locale-specific glyphs, ambient locale',
+    'Material3 - RichText TextSpan styles with different locales',
+    (WidgetTester tester) async {
+
+      await tester.pumpWidget(
+        MaterialApp(
+          theme: ThemeData(useMaterial3: false),
```
Comment (QuncCccccc) [inline packages/flutter/test/rendering/localized_fonts_test.dart:69]: Should here be `useMaterial3: true`?

File: packages/flutter/test/widgets/page_route_builder_test.dart [hunk 2]
```diff
@@ -91,13 +93,23 @@ class ModalPage extends StatelessWidget {
...
+    await tester.pumpWidget(const TestPage(useMaterial3: false));
+    await tester.tap(find.byType(FloatingActionButton));
+    await tester.pumpAndSettle();
+    await expectLater(
+      find.byType(TestPage),
+      matchesGoldenFile('m2_page_route_builder.barrier.png'),
+    );
+  });
+
   testWidgets('Barriers show when using PageRouteBuilder', (WidgetTester tester) async {
```
Comment (QuncCccccc) [inline packages/flutter/test/widgets/page_route_builder_test.dart:106]: nit: Should we here also show 'Material3 - ' at the beginning?

File: packages/flutter/test/rendering/localized_fonts_test.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/test/rendering/localized_fonts_test.dart]: Yes, good catch!

File: packages/flutter/test/widgets/page_route_builder_test.dart (hunk 2 above)
Comment (HansMuller) [reply packages/flutter/test/widgets/page_route_builder_test.dart]: Oops, yes this one should be labeled Material3.

**Review by QuncCccccc — APPROVED** [review]
LGTM:)

**Pull request merged**